### PR TITLE
Enhance the creation of qemu machine dev

### PR DESCRIPTION
**Pull request by luckyh**
* qcontainer: refactor the routine of `machine_by_params`
* qdevices: introduce `QMachine` and apply it to virttest

Signed-off-by: Xu Han <xuhan@redhat.com>

### CODE REVIEW

**Review by hereischen — APPROVED** [review]

**Comment by luckyh** [comment]
Hi @YongxueHong @PaulYuuu @zhencliu would you mind to help review this PR? thanks!

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1517,9 +1513,9 @@ def sort_key(dev):
                 pcics.sort(key=sort_key, reverse=False)
             devices.insert(pcics)
 
-        def add_secure_guest_object(params):
+        def add_secure_guest(params):
```
Comment (PaulYuuu) [inline virttest/qemu_vm.py:1516]: I would suggest adding some suffixes, `add_secure_guest` make me so confused, like adding a real guest.

**Comment by zhencliu** [comment]
LGTM for the secure guest part, thanks for the update.

File: virttest/qemu_vm.py (hunk 1 above)
Comment (luckyh) [reply virttest/qemu_vm.py]: Yes, make sense, have updated the method name to `add_secure_guest_descriptor`.

**Comment by luckyh** [comment]
Rebased to resolved the CI failures.

**Review by PaulYuuu — APPROVED** [review]

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -1433,125 +1412,81 @@ def machine_riscv64_mmio(cmd=False):
...
+        avocado_machine = ''
+        invalid_machine = None
+        if not machine_type:
+            for m_name, m_desc in self.__machines_info.items():
+                if '(default)' in m_desc:
+                    machine_type = m_name
+                    break
             else:
+                if machine_type is None:
+                    machine_type = "<unspecified>"
```
Comment (YongxueHong) [inline virttest/qemu_devices/qcontainer.py:1453]:
```suggestion
                machine_type = "<unspecified>"
```

**Review by YongxueHong — APPROVED** [review]
LGTM.

File: virttest/qemu_devices/qcontainer.py (hunk 2 above)
Comment (luckyh) [reply virttest/qemu_devices/qcontainer.py:1453]: Okay, the reason of adding this check is that we may want to distinguish empty strings as the value from the case by not given the param at all, to give users the right hint at the following warning message. We could discuss if needed, thanks!

Comment (YongxueHong) [reply virttest/qemu_devices/qcontainer.py:1453]: > Okay, the reason of adding this check is that we may want to distinguish empty strings as the value from the case by not given the param at all, to give users the right hint at the following warning message. We could discuss if needed, thanks!

All right, It looks good to me.
Thanks.

**Comment by YongxueHong** [comment]
Hi, all 
Thanks all, let's merge it.

**Pull request merged**